mne/preprocessing/ica.py

```diff
@@ -270,6 +271,22 @@ def __repr__(self):
 
         return '<ICA  |  %s>' % s
 
+    def _check_for_unsupported_ica_channels(self, picks, info):
+        """Check for channels in picks that are not considered
+        valid channels. Accepted channels are the data channels
+        ('seeg','ecog','eeg','mag', and 'grad') and 'eog'.
+        This prevents the program from crashing without
+        feedback when a bad channel is provided to ICA whitening.
+        """
+        types = _DATA_CH_TYPES_SPLIT + ['eog']
+        check = all([channel_type(info, j) in types for j in picks])
```

Comment:
Why is the list being sorted? Is it better to do this through sets than checking every list element?